frontend/src/AlertConfiguration/AlertConfigurationComponents/ConfigForm.js

```diff
@@ -1,49 +1,111 @@
-import React, { useState, Fragment } from 'react';
-import AppBar from '@material-ui/core/AppBar';
-import CssBaseline from '@material-ui/core/CssBaseline';
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import InputLabel from '@material-ui/core/InputLabel';
 import { makeStyles } from '@material-ui/core/styles';
-import Toolbar from '@material-ui/core/Toolbar';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import uuid from "uuid";
 import Typography from '@material-ui/core/Typography';
-import ControlPanel from './ControlPanel';
-import Footer from './Footer';
-import ConfigList from './ConfigList';
-
-const useStyles = makeStyles((theme) => ({
-  main: {
-    display: 'flex',
-    flexDirection: 'column',
-    minHeight: '100vh',
+import Tooltip from '@material-ui/core/Tooltip';
+
+const useStyles = makeStyles({
+  formControl: {
+    backgroundColor: 'white',
+    marginBottom: 25,
+    minWidth: 250,
   },
-}));
+});
+
 
-export default function AlertConfiguration() {
+export default function ConfigForm({ addConfig }) {
   const classes = useStyles();
 
-  const [configs, setConfigs] = useState([]);
+  const [config, setConfig] = useState({
+    id: '',
+    data: '',
+    rData: '',
+  });
+
+  {/**TODO: Replace drop-down menu and hard-coded data options with text-input that recommends data options*/}
+  const dimensionsMetrics = ["noodle", "spice", "egg", "soup", "instant noodle"];
+  const rDimensionsMetrics = ["noodle", "spice", "egg", "soup", "instant noodle"];
+
+  const dataOptions = (
+    dimensionsMetrics.map((d) => <MenuItem value={d}>{d}</MenuItem>)
+  );
+
+  const rDataOptions = (
+    rDimensionsMetrics.map((rd) => <MenuItem value={rd}>{rd}</MenuItem>)
+  );
+
+  function handleDataChange(event) {
+    setConfig({ ...config, data: event.target.value });
+  };
+
+  function handleRDataChange(event) {
+    setConfig({ ...config, rData: event.target.value });
+  };
 
-  function addConfig(config) {
-    setConfigs([config, ...configs]);
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (config.data.trim() && config.rData.trim()) {
+      addConfig({ ...config, id: 10 });
```

Comment:
is this id hardcoded?

remove 'magic numbers' and have it as some constant variable (for example DEFAULT_ID or something named similarly)

Reply:
Yes, the ID is hard-coded. I put a to-do to create unique IDs. I will take your advice of using a variable named DEFAULT_ID instead

Reply:
Completed